src/plugins/kubectlGeneration/aksDocsRag/client.ts

```diff
@@ -0,0 +1,114 @@
+import { ReadyAzureSessionProvider } from "../../../auth/types";
+import { Errorable, failed, getErrorMessage } from "../../../commands/utils/errorable";
+import { aksDocsRAGEndpoint, aksDocsRAGIntent, aksDocsRAGScenario, aksDocsRAGScopes } from "./constants";
+
+export function getAKSDocsRAGClient(sessionProvider: ReadyAzureSessionProvider) {
+    return new AKSDocsRAGClient(sessionProvider);
+}
+
+type RequestConfig = {
+    message: string;
+};
+
+export interface CommandResponse {
+    status: string;
+    message: string;
+    code: string;
+}
+
+export class AKSDocsRAGClient {
+    private authToken: string;
+    private tokenExpiresAt: number;
+    private sessionProvider: ReadyAzureSessionProvider;
+    private authScopes: string[];
+
+    constructor(sessionProvider: ReadyAzureSessionProvider) {
+        this.authToken = "";
+        this.tokenExpiresAt = 0;
+        this.sessionProvider = sessionProvider
+        this.authScopes = aksDocsRAGScopes;
+    }
+
+    private async fetchToken() {
+        const token = await this.sessionProvider.getAuthSession({ scopes: this.authScopes });
+
+        if ((!token || failed(token))) {
+            throw new Error(`No Microsoft authentication session found: ${token.error}`);
+        }
+
+        this.authToken = token.result.accessToken;
+    }
+
+    protected async refreshToken() {
+        if (!this.authToken || this.tokenExpiresAt < Date.now()) {
+            await this.fetchToken();
+        }
+    }
+
+    public async sendRequest(config: RequestConfig): Promise<Errorable<{ response: CommandResponse }>> {
+        await this.refreshToken();
+        const { message } = config;
+
+        let rawResponse = undefined;
+        try {
+            rawResponse = await fetch(aksDocsRAGEndpoint, {
+                method: "POST",
+                headers: {
+                    "Access-Control-Allow-Origin": "*",
+                    Authorization: `Bearer ${this.authToken}`,
+                    "App-Tenant-Id": this.sessionProvider.selectedTenant.id,
+                },
+                body: JSON.stringify({
+                    IsLocalEvaluation: false,
+                    Question: message,
+                    Scenario: aksDocsRAGScenario,
+                    Intent: aksDocsRAGIntent,
+                }),
+            });
+        } catch (err: unknown) {
+            return { succeeded: false, error: `Unable to fetch data from AKS Docs RAG endpoint: ${getErrorMessage(err)}` };
+        }
+
+        if (!rawResponse) {
+            return { succeeded: false, error: `Unable to fetch data from AKS Docs RAG endpoint` };
+        }
+
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
```

Comment:
💡 Please avoid this anywhere in code: `// eslint-disable-next-line @typescript-eslint/no-explicit-any` by doing this we are disabling listing checks for types, which could lead to variable type check.

Reply:
Updated